includes/Admin/Admin_AJAX.php

```diff
@@ -62,55 +63,36 @@ public function get_checks_to_run() {
 		$checks = filter_input( INPUT_POST, 'checks', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
 		$plugin = filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_STRING );
 
-		// Attempt to get the plugin basename based on the request.
-		try {
-			$plugin_basename = Plugin_Request_Utility::get_plugin_basename_from_input( $plugin );
-		} catch ( Exception $error ) {
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			$runner = new AJAX_Runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
 			wp_send_json_error(
-				new WP_Error( 'invalid-plugin', $error->getMessage() ),
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
```

Comment:
```suggestion
				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ), array( 'status' => 403 ) ),
```

Nitpic, you can pass status to WP_Error object.

Reply:
Same 
https://github.com/10up/plugin-check/pull/100/files#diff-9664c81d5e62051fb173b90454641ac9013c9397129f7e6adbb75f97978cad20R88

Reply:
Thanks @spacedmonkey as you've identified we've been consistent with the approach here so this change would require refactoring across multiple instances. With the current goal being to complete the architecture ahead of the milestone 1 review that starts next week, I would rather discuss and address changes like this then.

Reply:
Anything you are not planning to fix in this PR, then add a comment in the code.

Reply:
@spacedmonkey This one I'm not sure about, what would be the benefit of passing the status code to `WP_Error`? I know the REST API uses that, but this is a simple JSON response in an AJAX request, so I don't think this would do anything?

Reply:
This is a nicpic. Ignore me.